### PR TITLE
SHOP-1: Create basic shop project

### PR DESCRIPTION
**Pull request by Wraan**

### CODE REVIEW

**Review by Swistak06 — APPROVED** [review]
Everything is ok.

File: src/main/java/com/shop/controller/HomeController.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import com.shop.model.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class HomeController {
+
+
```
Comment (polskikiel) [inline src/main/java/com/shop/controller/HomeController.java]: unnecessary space

File: src/main/java/com/shop/model/User.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+
+@Entity
+@Table(name = "users")
+public class User {
+
+    @Id
+    @GeneratedValue
+    @Column(name = "user_id", unique = true, nullable = false)
```
Comment (polskikiel) [inline src/main/java/com/shop/model/User.java:13]: unnecessary unique and nullable properties if it's already @GenerateValue

File: src/main/java/com/shop/model/User.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    private String username;
+
+    @NotBlank
+    private String password;
+
+    public User() {
+    }
+
+    public User(String username, String password) {
+
```
Comment (polskikiel) [inline src/main/java/com/shop/model/User.java:26]: delete space

File: target/classes/application.properties [hunk 4]
```diff
@@ -0,0 +1,11 @@
+server.port=8081
```
Comment (polskikiel) [inline target/classes/application.properties:1]: shouldn't we have target/ dir in gitignore?

File: target/classes/application.properties [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+spring.datasource.driver-class-name=com.mysql.jdbc.Driver
+spring.datasource.url=jdbc:mysql://localhost:3306/shop?useSSL=false
+spring.datasource.username=root
+spring.datasource.password=password
+
+spring.jpa.hibernate.ddl-auto=update
```
Comment (polskikiel) [inline target/classes/application.properties:8]: I think that create-drop strategy would be better for developing

File: src/main/resources/application.properties [hunk 6]
```diff
@@ -0,0 +1,11 @@
+server.port=8081
+
+spring.datasource.driver-class-name=com.mysql.jdbc.Driver
+spring.datasource.url=jdbc:mysql://localhost:3306/shop?useSSL=false
+spring.datasource.username=root
+spring.datasource.password=password
+
+spring.jpa.hibernate.ddl-auto=update
+
+spring.servlet.multipart.max-file-size=10MB
```
Comment (polskikiel) [inline src/main/resources/application.properties:10]: do we need these multipart properties in that PR?

File: src/main/java/com/shop/model/User.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+@Entity
+@Table(name = "users")
+public class User {
+
+    @Id
+    @GeneratedValue
+    @Column(name = "user_id", unique = true, nullable = false)
+    private long id;
+
+    @NotBlank
```
Comment (polskikiel) [inline src/main/java/com/shop/model/User.java:16]: Not needed in that PR

File: shop.iml [hunk 8]
```diff
@@ -0,0 +1,120 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (polskikiel) [inline shop.iml:1]: do we need that file? ;p

**Review by polskikiel — CHANGES_REQUESTED** [review]
please do only the things which are necessary in PR, otherwise code getting dirty

**Pull request closed without merging**